segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -408,5 +441,10 @@ private void logError(String segment, Throwable u) {
         }
     }
 
+    @Override
+    public void close() {
+        // Todo: Delete Transient Segments.
```

Comment:
This is a good step however it may not cover all cases. If the connection is correctly closed (for whatever reason), this may send a signal to delete the transient segment. However:
1. There is no guarantee that the "delete" will actually make it to the segment container and be processed.
2. The process may crash and this may not even be invoked at all.

It is for this reason that I wanted to leave this mechanism for a subsequent PR; we'd have to think about a "clean" way to check if a transient segment is still used from within the segment store.